test/integration/retryable-writes/retryable_writes.spec.test.ts

```diff
@@ -150,8 +161,17 @@ async function executeScenarioTest(test, ctx: RetryableWriteTestContext) {
     const result = resultOrError;
     const expected = test.outcome.result;
 
-    // TODO(NODE-4034): Make CRUD results spec compliant
-    expect(result.value ?? result).to.deep.include(expected);
+    const actual = result.value ?? result;
+    // Some of our result classes contain the optional 'acknowledged' property which is
+    // not part of the test expectations.
+    for (const property in expected) {
+      expect(actual).to.have.deep.property(property, expected[property]);
+    }
+
+    // Check we don't have any extra properties on the bulk write result.
```

Comment:
Instead of whitelisting the allowed properties, we could compare the properties on `actual` to those on `expected` and assert there are no extra properties

Reply:
Actually this I've just seen as a problem to assert as the result on `findOneAnd...` methods is not a `BulkWriteResult` but the actual matching document - so it can have any properties on it. So going to remove the check on L171.